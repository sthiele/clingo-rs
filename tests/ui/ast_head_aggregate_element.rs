use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let mut term1 = ast::Term::from(sym);
    let mut term2 = ast::Term::from(sym);
    let tuple = vec![term1];
    let lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
    let lit2 = ast::Literal::from_term(ast::Sign::NoSign, &term2);
    let condition = vec![lit2];
    let cond = ast::ConditionalLiteral::new(&lit, &condition);
    let helem = ast::HeadAggregateElement::new(&tuple,cond);
    
    term1 =  ast::Term::from(sym);
    term2 =  ast::Term::from(sym);
    lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
    drop(tuple);
    let _end = (term1, term2, lit, helem);
}