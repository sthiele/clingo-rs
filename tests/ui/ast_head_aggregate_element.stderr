error[E0506]: cannot assign to `term1` because it is borrowed
  --> $DIR/ast_head_aggregate_element.rs:14:5
   |
8  |     let lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
   |                                                          ------ borrow of `term1` occurs here
...
14 |     term1 =  ast::Term::from(sym);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `term1` occurs here
...
18 |     let _end = (term1, term2, lit, helem);
   |                                    ----- borrow later used here

error[E0506]: cannot assign to `term2` because it is borrowed
  --> $DIR/ast_head_aggregate_element.rs:15:5
   |
9  |     let lit2 = ast::Literal::from_term(ast::Sign::NoSign, &term2);
   |                                                           ------ borrow of `term2` occurs here
...
15 |     term2 =  ast::Term::from(sym);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `term2` occurs here
...
18 |     let _end = (term1, term2, lit, helem);
   |                                    ----- borrow later used here

error[E0384]: cannot assign twice to immutable variable `lit`
  --> $DIR/ast_head_aggregate_element.rs:16:5
   |
8  |     let lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
   |         ---
   |         |
   |         first assignment to `lit`
   |         help: make this binding mutable: `mut lit`
...
16 |     lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot assign twice to immutable variable

error[E0505]: cannot move out of `tuple` because it is borrowed
  --> $DIR/ast_head_aggregate_element.rs:17:10
   |
12 |     let helem = ast::HeadAggregateElement::new(&tuple,cond);
   |                                                ------ borrow of `tuple` occurs here
...
17 |     drop(tuple);
   |          ^^^^^ move out of `tuple` occurs here
18 |     let _end = (term1, term2, lit, helem);
   |                                    ----- borrow later used here
